extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudParsingHelper.java

```diff
@@ -145,12 +145,15 @@ public static String resolveUrlWithEmbedPlayer(String apiUrl) throws IOException
      *
      * @return the resolved id
      */
-    public static String resolveIdWithWidgetApi(String urlString) throws IOException, ReCaptchaException, ParsingException {
+    public static String resolveIdWithWidgetApi(String urlString) throws IOException, ParsingException {
         // Remove the tailing slash from URLs due to issues with the SoundCloud API
-        if (urlString.charAt(urlString.length() - 1) == '/') urlString = urlString.substring(0, urlString.length() - 1);
-        // Make URL lower case and remove www. if it exists.
+        if (urlString.charAt(urlString.length() - 1) == '/') {
+            urlString = urlString.substring(0, urlString.length() - 1);
+        }
+        // Ensure the host of the URL equals "soundcloud.com" and the URL is lowercase.
         // Without doing this, the widget API does not recognize the URL.
-        urlString = Utils.removeWWWFromUrl(urlString.toLowerCase());
+        urlString = Utils.replaceHttpWithHttps(Utils.removeWWWFromUrl(urlString.toLowerCase()))
+                .replaceFirst("m\\.soundcloud\\.com", "soundcloud.com");
```

Comment:
I did not want to replace the whole host, in case there are different requests, too. But maybe, that is a cleaner way to do it.

Reply:
On my side, I am working on some improvements in SoundCloud and I created a string M_PATTERN which parses only m. URLs and renamed the `removeWWWFromUrl` method of the Utils class to `removeMAndWWWFromUrl`:

```java
    private static final Pattern M_PATTERN = Pattern.compile("(https?)?:\\/\\/m\\.");

    [...]

    public static String removeMAndWWWFromUrl(final String url) {
        if (M_PATTERN.matcher(url).find()) {
            return url.replace("m.", "");
        }
        if (WWW_PATTERN.matcher(url).find()) {
            return url.replace("www.", "");
        }
        return url;
    }
```

What do you think?

Reply:
looks better! feel free to push to this PR

Reply:
Done: #627.

Reply:
Eh? @TiA4f8R By pushing to this PR, he meant that you could make changes to this branch freely, not that you should create a new branch and make a new PR.

Or was there a reason you did that?

Reply:
My bad, I didn't read carefully the message. However, the reason why I used a new branch and created a new PR is that I created locally the changes before this PR (and commited them). Also, my PR does more changes than this PR.